Installing privately hosted packages from VCS fails under 7.0.0

We use PIP to install packages from private VCS systems.  After upgrading to 7.0.0 installs started failing with: 

```
AssertionError: nameofpackage .dist-info directory not found
```

While this seems to be resolved by just opting out of wheels ```--no-use-wheels```. I wanted to make sure this was the desired behavior.

--- a/pip/index.py
+++ b/pip/index.py
@@ -1207,6 +1207,19 @@ class Link(object):
     def is_wheel(self):
         return self.ext == wheel_ext
 
+    @property
+    def is_artifact(self):
+        """
+        Determines if this points to an actual artifact (e.g. a tarball) or if
+        it points to an "abstract" thing like a path or a VCS location.
+        """
+        from pip.vcs import vcs
+
+        if self.scheme in vcs.all_schemes:
+            return False
+
+        return True
+
 
 # An object to represent the "link" for the installed version of a requirement.
 # Using Inf as the url makes it sort higher.

--- a/pip/wheel.py
+++ b/pip/wheel.py
@@ -110,6 +110,8 @@ def cached_wheel(cache_dir, link, format_control, package_name):
         return link
     if link.is_wheel:
         return link
+    if not link.is_artifact:
+        return link
     if not package_name:
         return link
     canonical_name = pkg_resources.safe_name(package_name).lower()
@@ -714,6 +716,8 @@ class WheelBuilder(object):
                     logger.info(
                         'Skipping bdist_wheel for %s, due to being editable',
                         req.name)
+            elif autobuilding and not req.link.is_artifact:
+                pass
             elif autobuilding and not req.source_dir:
                 pass
             else:

